copier/template.py

```diff
@@ -426,11 +427,20 @@ def templates_suffix(self) -> str:
     def local_abspath(self) -> Path:
         """Get the absolute path to the template on disk.
 
-        This may clone it if `url` points to a
-        VCS-tracked template.
+        This may clone it if `url` points to a VCS-tracked template.
+        Dirty changes for local VCS-tracked templates will be copied.
         """
         result = Path(self.url)
         if self.vcs == "git":
+            if result.is_dir():
+                with local.cwd(result):
+                    if bool(git("status", "--porcelain").strip()):
+                        git("add", ".")
```

Comment:
Do we want `-A` as well? Without it, removed files would not be added, leaving some dirtiness.

Reply:
Yep, good catch. Although removed files are currently not staged anyway during update—looks like the same issue as in #461. I took a stab fixing it by removing files that were present in `dst` and not `src`, but this is difficult when you factor in templating and it was causing errors in other tests.